src/main.js

```diff
@@ -1,175 +1,95 @@
-    routeProxyObj.proxyConfig.auth = `${routeDestination.User}:${routeDestination.Password}`;
-  }
-
-  return routeProxyObj;
-}
+        if (debugMode) {
+          logger.info(`Serving ${req.url} from ${target}`);
+        }
 
-function _getFormattedUrlForResource (sResourceUrl) {
-  var resourceRoute = routesConfig["/resources"].target,
-      rResourcePattern = new RegExp(`/resources(/.*)$`);
-  
-  sResourceUrl = `${resourceRoute.version ? '/' + resourceRoute.version : ''}${resourceRoute.entryPath || "/resources"}${rResourcePattern.exec(sResourceUrl)[1]}`;
-  return sResourceUrl;
-}
+        serve(req, res, finalhandler(req, res));
 
-/**
- * Custom UI5 Server middleware example
- *
- * @param {Object} parameters Parameters
- * @param {Object} parameters.resources Resource collections
- * @param {module:@ui5/fs.AbstractReader} parameters.resources.all Reader or Collection to read resources of the
- *                                        root project and its dependencies
- * @param {module:@ui5/fs.AbstractReader} parameters.resources.rootProject Reader or Collection to read resources of
- *                                        the project the server is started in
- * @param {module:@ui5/fs.AbstractReader} parameters.resources.dependencies Reader or Collection to read resources of
- *                                        the projects dependencies
- * @param {Object} parameters.options Options
- * @param {string} [parameters.options.configuration] Custom server middleware configuration if given in ui5.yaml
- * @param {string} [parameters.options.configuration.destination_path] Absolute Path in local system to find the set of destinations file
- * @param {string} [parameters.options.configuration.debug] If set, Logs messages for which calls are being proxied & from where
- * @returns {function} Middleware function to use
- */
-function createMiddleware({resources, options}) {
-  destinations = require(options.configuration.destination_path)["destinations"];
-  routes = require(resources.rootProject._readers[0]._project.path + "/neo-app.json")["routes"];
-  isDebugMode = options.configuration.debug || false;
-
-  initProxyDestination(); //Build all the configs for upcoming requests
-
-  return async function proxyDestinations(req, res, next) {
-    try {
-      let matchedRoute;
-      
-      // Add headers for fetch
-      addHeader(res, HEADER_ALLOW_CONTROL_ALLOW_ORIGIN, "*");
-      addHeader(res, HEADER_ALLOW_CONTROL_ALLOW_CREDENTIALS, "true");
-      addHeader(res, HEADER_ALLOW_CONTROL_ALLOW_METHODS, "GET,PUT,POST,DELETE");
-      addHeader(res, HEADER_ALLOW_CONTROL_ALLOW_HEADERS, "X-Requested-With, Accept, Origin, Referer, User-Agent, Content-Type, Authorization, X-Mindflash-SessionID");
-      // Intercept OPTIONS method
-      if (req.method === "OPTIONS") {
-        // addHeader(res, HEADER_ALLOW_CONTROL_ALLOW_ORIGIN, "*");
-        res.status(200);
-        next();
-        return;
-      }
+      } else if (serveWithProxy) {
```

Comment:
There is a use-case where if user doesn't provide any configuration for SAPUI5 resources in ui5.yml . None of the if/elseif condition will proxy for **route** of _type_ **service**.